.github/workflows/ci.yml

```
@@ -21,11 +21,14 @@ jobs:
   fmt:
     name: Rustfmt
     runs-on: ubuntu-latest
+    timeout-minutes: 10
```

What benefit do timeouts give?

---

Timeouts prevent runaway jobs from consuming runner resources and exceeding GitHub Actions limits.

**From GitHub Actions documentation** ([jobs.<job_id>.timeout-minutes](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes)):
> "The maximum number of minutes to let a job run before GitHub automatically cancels it. **Default: 360**"

**Benefits:**
- **Resource efficiency**: Prevents jobs from hanging indefinitely if a command stalls, freeing up runner capacity for other projects
- **Fast failure detection**: A 10-minute timeout on `cargo fmt --check` that normally takes seconds will fail quickly if something goes wrong, rather than waiting the full 360-minute default
- **Explicit expectations**: Documents expected runtime, making performance regressions obvious
- **Responsible resource use**: Reduces unnecessary compute usage (better for the environment and GitHub's infrastructure)

For the `fmt` and `custom-checks` jobs, 10 minutes is generous (they typically take <1 minute), but provides a reasonable safety net while still failing much faster than the 6-hour default.